WordPress/gutenberg
# Add Inline Error for Protected Post Password Field and Uncheck "Password Protected" for Empty Field

Fixes: #64157

## What?
This PR addresses two issues related to the post password field:
1. Adds an inline error to warn users when a password exceeds the 255-character limit, preventing server errors.
2. Automatically unchecks the "Password Protected" checkbox when the password field is cleared, ensuring the UI accurately reflects the state of password protection.

## Why?

1. The inline error prevents users from submitting passwords that are too long and helps avoid server errors. It also improves the user experience by providing clear feedback when the password is truncated.
2. Unchecking the `Password Protected` checkbox when the password field is cleared ensures that the UI is in sync with the actual post protection state, reducing confusion for users.

## How?

1. For the inline error, the code checks the length of the entered password. If the length exceeds 255 characters, an inline error message is displayed.
2. To fix the `Password Protected` checkbox, the state is updated to automatically uncheck the box when the password field is cleared. This is done by updating the `showPassword` state when the password field is empty & the `Dropdown` is out of `Focus`, ensuring the checkbox reflects the correct state.

## Testing Instructions

1. Test Inline Error for Password Length:
    - Go to the post editor and enter a password longer than 255 characters in the `Password` field.
    - An inline error should appear, notifying the user that the password exceeds the character limit and will be truncated.

2. Test `Password Protected` Checkbox Behavior:
    - Create or edit a password-protected post.
    - Check the `Password Protected` checkbox and enter a password.
    - Save the post, then clear the password field.
    - Ensure that the "Password Protected" checkbox is automatically unchecked when the password field is cleared.
    - Save the post again and confirm that the password protection is disabled.

## Screenshots or screencast 



https://github.com/user-attachments/assets/60adb4e7-6b4d-4ad1-bfe4-4a250cf16bce
